src/System.Management.Automation/engine/hostifaces/PSTask.cs

```diff
@@ -763,18 +802,28 @@ public bool Add(PSTaskChildJob childJob)
         /// </summary>
         public void StopAll()
         {
+            _stopping = true;
+
             // Accept no more input
             Close();
             _stopAll.Set();
 
             // Stop all running tasks
+            PSTaskBase[] tasksToStop;
             lock (_syncObject)
             {
-                foreach (var task in _taskPool.Values)
-                {
-                    task.Dispose();
-                }
+                tasksToStop = new PSTaskBase[_taskPool.Values.Count];
+                _taskPool.Values.CopyTo(tasksToStop, 0);
```

Comment:
I'm curious why this copy is needed. It's a reference copy, right? So both `tasksToStop` and `_taskPool` will pointing to the same set of `PSTaskBase` objects.